libdnf5/base/log_event.cpp

```diff
@@ -153,23 +153,52 @@ std::string LogEvent::to_string(
                 } else {
                     return ret.append(utils::sformat(_("No match for group package: {}"), *spec));
                 }
+            } else if (goal_action_is_replay(action)) {
+                return ret.append(
+                    utils::sformat(_("Cannot perform {}, no match for: {}."), goal_action_to_string(action), *spec));
             }
             return ret.append(utils::sformat(_("No match for argument: {}"), *spec));
         case GoalProblem::NOT_FOUND_IN_REPOSITORIES:
             return ret.append(utils::sformat(
                 _("No match for argument '{0}' in repositories '{1}'"),
                 *spec,
                 utils::string::join(settings->get_to_repo_ids(), ", ")));
-        case GoalProblem::NOT_INSTALLED:
+        case GoalProblem::NOT_INSTALLED: {
+            if (goal_action_is_replay(action)) {
+                return ret.append(utils::sformat(
+                    _("Cannot perform {} for {} '{}' becasue it is not installed."),
```

Comment:
May I ask you to mark an order of arguments `{1}`. It will allow to change order for translators. In this case what about to provide a context message for translators?

Reply:
Added the marks but I don't know what are context messages? Can we perhaps add them separately?

Reply:
It is used to explain what message means. The method is `C_(<message for translaters>, <string to translate>)` with two string arguments and in DNF5 it is used only here - https://github.com/rpm-software-management/dnf5/blob/3fa8d63db9c060e1744cfd268cbba918941d2fd0/libdnf5/base/transaction.cpp#L1373.